src/Runner.jl

```diff
@@ -8,12 +8,14 @@ const AnyRedirectable = Union{Base.AbstractCmd, Base.TTY, IOStream, IOBuffer}
 # Host platform _must_ match the C++ string ABI of the binaries we get from the
 # repositories.  Note: when preferred_gcc_version=v"4" we can't really build for
 # that C++ string ABI :-(
+# Non-Linux platforms *should* end up using DockerRunner
+# so we can safely assume x86_64 support
 """
     default_host_platform
 
 The default host platform in the build environment.
 """
-const default_host_platform = Platform(arch(HostPlatform()), "linux"; libc="musl", cxxstring_abi="cxx11")
+const default_host_platform = Sys.islinux(HostPlatform()) ? Platform(arch(HostPlatform()), "linux"; libc="musl", cxxstring_abi="cxx11") : Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11")
```

Comment:
I'd simply do
```suggestion
const default_host_platform = Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11")
```
because for the foreseeable future we aren't going to change it anyway.  _If_ and when we'll have an extra host we'll figure out a mechanism to choose it.